kratos/python/containers_interface.h

```diff
@@ -71,7 +71,7 @@ class PointerVectorSetPythonInterface
         .def("__setitem__",  [](TContainerType& self, typename TContainerType::value_type& value){self[value.Id()] = value;} ) 
         .def("__setitem__",  [](TContainerType& self, typename TContainerType::pointer& pvalue){self(pvalue->Id()) = pvalue;} ) 
         .def("__getitem__",  [](TContainerType& self, unsigned int i){return self(i);} )
-        .def("__iter__",     [](TContainerType& self){return make_iterator(self.ptr_begin(), self.ptr_end());},  keep_alive<0,1>())  //TODO: decide if here we should use ptr_iterators or iterators
+        .def("__iter__",     [](TContainerType& self){return make_iterator(self.begin(), self.end());},  keep_alive<0,1>())  //TODO: decide if here we should use ptr_iterators or iterators
```

Comment:
Why you need to use the keep_alive here? I imagine that it won't hurt but I'm afraid if we miss the fact that the pointee is an internal reference.

Reply:
it simply does not work, even a minimal testcase if you don't ... the keep_alive is used in ALL the pybind tests...

Reply:
I digged a bit and I THINK that iterators give back an internal reference by default. that is, 

     make_iterator(...)

 is in my understanding equivalent to

     make_iterator<return_value_policy::reference_internal>(...)

from an experimental point of view, if i explicitly add the template parameter it does work, although with no visible change.

**if i remove the keep_alive it fails**